File: app/src/main/java/com/lkms/data/model/java/Booking.java
package com.lkms.data.model.java;

import com.google.gson.annotations.SerializedName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Đại diện cho một hàng trong bảng "Booking".
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Booking {

    @SerializedName("bookingId")
    private Integer bookingId;

    // Khóa ngoại đến User
    @SerializedName("userId")
    private Integer userId;

    // Khóa ngoại đến Equipment
    @SerializedName("equipmentId")
    private Integer equipmentId;

    // Khóa ngoại đến Experiment
    @SerializedName("experimentId")
    private Integer experimentId;

    @SerializedName("startTime")
    private String startTime;

    @SerializedName("endTime")
    private String endTime;

    @SerializedName("bookingStatus")
    private String bookingStatus;

    @SerializedName("rejectReason")
    private String rejectReason;

    @SerializedName("equipmentName")
    private String equipmentName;
}

File: app/src/main/java/com/lkms/data/model/Booking.kt
// Trong file: data/model/Booking.kt
package com.lkms.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Đại diện cho một hàng trong bảng "Booking".
 */
@Serializable
data class Booking(
    @SerialName("bookingId")
    val bookingId: Int? = null,

    // Khóa ngoại đến User
    @SerialName("userId")
    val userId: Int? = null,

    // Khóa ngoại đến Equipment
    @SerialName("equipmentId")
    val equipmentId: Int? = null,

    // Khóa ngoại đến Experiment
    @SerialName("experimentId")
    val experimentId: Int? = null,

    @SerialName("startTime")
    val startTime: String? = null,

    @SerialName("endTime")
    val endTime: String? = null,

    @SerialName("bookingStatus")
    val bookingStatus: String? = null,

    @SerialName("rejectReason")
    val rejectReason: String? = null
)